app/components/finance/statements/ecf_details_table.rb

```diff
@@ -9,7 +9,7 @@ class ECFDetailsTable < BaseComponent
 
       def initialize(statement:)
         @statement = statement
-        @calculator = Finance::ECF::StatementCalculator.new(statement: @statement)
+        @calculator = Finance::ECF::ECT::StatementCalculator.new(statement: @statement)
```

Comment:
I think the old calculators will support ECF in the old world and ECTs in the new world. Maybe we can keep them as is with ECF instead? and pass in the class in the initialiser instead, and default to both otherwise?

Reply:
this might help us avoiding `Finance::ECF::ECT`, and instead we'll have `Finance::ECF` for old world and ECTs in new world, but controlled by passing in the class to the calculator, let me know your thoughts